core/common/src/main/java/alluxio/security/LoginUser.java

```diff
@@ -50,18 +50,21 @@ private LoginUser() {} // prevent instantiation
    * @return the login user
    * @throws IOException if login fails
    */
-  public static User get() throws IOException {
+  public static synchronized User get() throws IOException {
```

Comment:
It will reduce performance, I don't think we call this enough for the difference to be noticeable. However, if we don't synchronize we will have a race condition. Without a memory barrier like synchronized, a thread could read stale or partial state from a LoginUser created by another thread. See http://www.javaworld.com/article/2074979/java-concurrency/double-checked-locking--clever--but-broken.html